moveit_simple/test/moveit_simple.cpp

```diff
@@ -50,50 +50,185 @@ TEST(MoveitSimpleTest, reachability)
   ASSERT_FALSE(robot.isReachable("waypoint4"));  //stored in the URDF
 }
 
+
 TEST(MoveitSimpleTest, add_trajectory)
 {
   moveit_simple::Robot robot(ros::NodeHandle(), "robot_description", "manipulator");
   const std::string TRAJECTORY_NAME("traj1");
   const Eigen::Affine3d pose = Eigen::Affine3d::Identity(); 
   ros::Duration(2.0).sleep();  //wait for tf tree to populate
+  const moveit_simple::InterpolationType cart = moveit_simple::CARTESIAN;
+  const moveit_simple::InterpolationType joint = moveit_simple::JOINT;
 
   ROS_INFO_STREAM("Testing loading of unknown point, should fail");
   EXPECT_THROW(robot.addTrajPoint("bad_traj", "unknown_name", 1.0),std::invalid_argument);
-  EXPECT_THROW(robot.addTrajPoint(TRAJECTORY_NAME, pose, "random_link", 5.0, "unknown_name"), tf2::TransformException);
-
+  EXPECT_THROW(robot.addTrajPoint(TRAJECTORY_NAME, pose, "random_link", 5.0), tf2::TransformException);
   ROS_INFO_STREAM("Testing trajectory adding of points");
   EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, "home",      0.5));
-  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, "waypoint1", 1.0));
-  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, "tf_pub1",   2.0, "CARTESIAN", 5));
+  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, "waypoint1", 1.0, joint, 5));
+  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, "tf_pub1",   2.0, cart, 8));
+//  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, "tf_pub1",   2.0));
   EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, "waypoint2", 3.0));
-  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, "waypoint3", 4.0));
-  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, pose, "tool0", 5.0, "random_name"));
+  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, "waypoint3", 4.0, cart));
+  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, pose, "tool0", 5.0));
   EXPECT_NO_THROW(robot.execute(TRAJECTORY_NAME));
 
   EXPECT_NO_THROW(robot.addTrajPoint("traj2", "waypoint4", 4.5));
   EXPECT_THROW(robot.execute("traj2"), moveit_simple::IKFailException);
 
-  EXPECT_NO_THROW(robot.addTrajPoint("traj3", "waypoint2", 1.0));
-  EXPECT_NO_THROW(robot.addTrajPoint("traj3", "waypoint3", 2.0, "CARTESIAN", 1));
-  EXPECT_THROW(robot.execute("traj3"), moveit_simple::IKFailException);
-
   EXPECT_THROW(robot.execute("bad_traj"), std::invalid_argument);
 
 }
 
 
+TEST(MoveitSimpleTest, interpolation)
+{
+  class Robot2: public moveit_simple::Robot
```

Comment:
Use a better name, such at `RobotTest`.  Also, add a comment that indicates why you created this class.